routing/router.go

```diff
@@ -1055,7 +1056,48 @@ func (r *ChannelRouter) SendPayment(payment *LightningPayment) ([32]byte, *Route
 		if sendError != nil {
 			log.Errorf("Attempt to send payment %x failed: %v",
 				payment.PaymentHash, sendError)
-			continue
+
+			switch sendError.(type) {
```

Comment:
At this point, the only errors that should case a complete halt in the payment attempt are: 
```go
lnwire.CodeInvalidRealm
lnwire.CodeInvalidOnionVersion
lnwire.CodeInvalidOnionKey
lnwire.CodeInvalidOnionHmac
lnwire.CodeRequiredChannelFeatureMissing
lnwire.CodeUnknownPaymentHash                     
lnwire.CodeIncorrectPaymentAmount                
lnwire.CodeFinalExpiryTooSoon           
lnwire.CodeFinalIncorrectCltvExpiry     
lnwire.CodeFinalIncorrectHtlcAmount     
lnwire.CodeAmountBelowMinimum                     
lnwire.CodeFeeInsufficient                       
lnwire.CodeIncorrectCltvExpiry                    
lnwire.CodeExpiryTooSoon                          
```